packages/@aws-cdk/aws-apigateway/lib/private/restapi-base.ts

```diff
@@ -0,0 +1,206 @@
+import * as iam from '@aws-cdk/aws-iam';
+import { CfnOutput, Construct, Resource, Stack } from '@aws-cdk/core';
+import { CfnAccount, CfnRestApi } from '../apigateway.generated';
+import { Deployment } from '../deployment';
+import { DomainName, DomainNameOptions } from '../domain-name';
+import { GatewayResponse, GatewayResponseOptions } from '../gateway-response';
+import { Method } from '../method';
+import { IResource } from '../resource';
+import { IRestApi, RestApiOptions } from '../restapi';
+import { Stage } from '../stage';
+import { UsagePlan, UsagePlanProps } from '../usage-plan';
+
+const RESTAPI_SYMBOL = Symbol.for('@aws-cdk/aws-apigateway.RestApiBase');
+
+/**
+ * @internal
+ */
+export abstract class RestApiBase extends Resource implements IRestApi {
```

Comment:
FYI - All of this class except for the `isRestApi()` method is a pure move from `restapi.ts`.